connectors/pdepend.py

```diff
@@ -0,0 +1,142 @@
+import logging
+import os
+import subprocess
+import tempfile
+from os.path import exists
+import xml.etree.ElementTree as ET
+
+from models.metric import Metric
+from utils.math import Math
+from utils.timeit import timeit
+
+
+class PDependConnector:
+    """
+    Connector to PHP Depend CLI tool
+    https://github.com/pdepend/pdepend
+
+    Attributes:
+    -----------
+        - directory   Full path to a cloned GIT repository
+        - session     Connection to a database managed by sqlalchemy
+        - version     Sqlalchemy object representing a Version
+    """
+    def __init__(self, directory, version, session, config):
+        self.directory = directory
+        self.session = session
+        self.version = version
+        self.configuration = config
+
+    def analyze_source_code(self):
+        """
+        Analyze the repository by using PHP Depend analysis tool
+        """
+        logging.info('PDepend::analyze_repo')
+        # Test if metrics have been already generated for this version
+        metric = self.session.query(Metric).filter(Metric.version_id == self.version.version_id).first()
+        if not metric:
+            metric = Metric()
+        if self.configuration.language != "PHP":
```

Comment:
don't need this

Reply:
As discussed, the connector now raises an exception if the language is not the right one. Added the same exception to JPeek, CK and Radon connectors.